### ISSUE TITLE
ClientCapabilities seem to be missing lenses

### ISSUE THREAD

**Issue by fwcd** [haskell/lsp#593]
I am currently migrating the Curry language server from `lsp` 1.6 to 2.3 and currently wonder how to migrate this code snippet:

```haskell
import qualified Language.LSP.Protocol.Types as J
import qualified Language.LSP.Protocol.Lens as J
import qualified Language.LSP.Protocol.Message as J

...

fromMaybe False (do
  docCapabilities <- capabilities ^. J.textDocument
  cmCapabilities <- docCapabilities ^. J.completion
  ciCapabilities <- cmCapabilities ^. J.completionItem
  ciCapabilities ^. J.snippetSupport)
```

where `capabilities :: ClientCapabilities`. The `snippetSupport` lens seems to be missing, at least I could not find it. Any ideas?

**Comment by michaelpj** [comment]
That seems like it should be okay. That code compiles for me. What version of the library are you using?

**Comment by michaelpj** [comment]
Oh, you said, 2.3. Then I'm confused :/

**Comment by fwcd** [comment]
Ah wait, I thought 2.3 was the newest version since that was what the recently released LTS 22.31 Stackage gave me. Might have to add 2.7 as an extra dependency and try if that fixes it.

**Comment by fwcd** [comment]
Ah I think part of what caused my confusion that the badges in the README all displayed 2.7 😄

Submitted #594 to fix it

**Comment by fwcd** [comment]
Seems to already have been implemented in lsp 2.7/lsp-types 2.3, therefore I'll close this.
